src/edu/csus/ecs/pc2/shadow/ShadowController.java

```diff
@@ -305,223 +304,198 @@ private IRemoteContestAPIAdapter createRemoteContestAPIAdapter(URL url, String l
 //            log.warning("Shadow Controller 'getJudgementComparisonInfo()' called when Shadow controller is not running"); 
```

Comment:
I'm not sure what "these" refers to.  However, I've already noted in several other comments that changes which appear at first glance not to be related to the NPE were in fact definitely related (for example, the need to change the background color in the grid because of the change in comparison logic).  Please see those other comments.